src/dune/file_tree.ml

```diff
@@ -387,6 +414,17 @@ end = struct
     let dune_file =
       dune_file ~dir_status ~recognize_jbuilder_projects ~files ~project ~path
     in
+    let dirs = List.map dirs ~f:(fun (n, s, f) -> (n, s, Some f)) in
+    let dirs =
+      match dune_file with
+      | None
+      | Some (Ocaml_script _) ->
+        dirs
+      | Some (Plain dune_file) ->
+        dirs
+        @ ( Sub_dirs.Dir_map.sub_dirs dune_file.for_subdirs
+          |> List.map ~f:(fun s -> (s, Path.Source.relative path s, None)) )
```

Comment:
We now have directories that don't exist in the source but do contain dune files. We can include them in the normal list of directories as I've done here, or we can store them in a separate field as well. @diml what do you think?